src/blas/backends/cublas/cublas_level1.cpp

```diff
@@ -477,15 +478,18 @@ inline void iamax(const char* func_name, Func func, sycl::queue& queue, int64_t
     queue.submit([&](sycl::handler& cgh) {
         auto int_res_acc = int_res_buff.template get_access<sycl::access::mode::read>(cgh);
         auto result_acc = result.template get_access<sycl::access::mode::write>(cgh);
-        cgh.single_task(
-            [=]() { result_acc[0] = std::max((int64_t)int_res_acc[0] - 1, (int64_t)0); });
+        cgh.single_task([=]() {
```

Comment:
I think it would be better to skip the `single_task` entirely if `base == index_base::one`, ie
```c++
if (base == oneapi::math::index_base::zero) {
    cgh.single_task(
                [=]() { result_acc[0] = std::max((int64_t)int_res_acc[0] - 1, (int64_t)0); });
}
```

Reply:
I think we cannot skip the entire kernel. Besides "-1" for `index_base::zero`, it also does `int` to `int64_t` casting and assigns `int_res_acc`(int) to `result_acc`(int64_t).